### DISCUSSION TITLE
Build the official openebs/charts as a chart manager for dependent component charts

### DISCUSSION THREAD

**Discussion by akhilerm** [openebs/openebs#3340] [Ideas] [answered]
Currently [openebs/charts](https://openebs.github.io/charts/) serves as the official way to install all the openebs components in a cluster. The values for each component (eg: NDM, localPV etc) is also maintained in the main charts repo. This makes it difficult to keep the individual components in sync.  To tackle this, openebs has started to create independent helm charts for its components which are available :
- [x] [CStor](https://openebs.github.io/cstor-operators/)
- [x] [Dynamic LocalPV](https://openebs.github.io/dynamic-localpv-provisioner/)
- [x] [ZFS LocalPV](https://openebs.github.io/zfs-localpv/)
- [ ] LVM LocalPV
- [ ] Jiva
- [x] [NDM](https://openebs.github.io/node-disk-manager/)

Going forward, the official openebs-charts should be able to install openebs from these dependent charts. This will make managing the changes easier and adding additional components much more flexible.

#### Advantages of the above approach
- easier managing of charts of all components which may sometimes follow its own release versioning (like NDM / ZFS LocalPV)
- ability to choose storage engine at time of installation and install only the necessary components. This will be a huge advantage as the current charts install all the components even if not required. After installation also, if the user wants another stoarge engine in the cluster, they can apply the new `values.yaml` and necessary components will get installed.
- Adding new storage engines become easier. This will make it easier to add Mayastor charts.
- Deprecate the old cstor scheme (SPC) in favour of the new scheme.

#### Disadvantages
- upgrading from the older version to a newer version with dependent cases has to be thoroughly thought through as there will be lot of changes regarding component labels and versions.

#### Goals
- [ ] Replace cstor(SPC) with cstor-operators chart
- [ ] Replace localPV with dynamic localPV charts
- [ ] Add ZFS localPV charts
- [ ] Replace jiva with jiva operator charts once it becomes available.

**Comment by avishnu** [comment, marked as answer]
Above has been achieved in the subsequent releases of the helm chart.
